src/components/StatesButton/StatesButton.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import style from './StatesButton.st.css';
+import { Button, SIZE } from '../Button';
+import { Check } from '../../icons/dist';
+
+const delay = time => new Promise(resolve => setTimeout(resolve, time));
+
+export interface StatesButtonProps {
+  className?: string;
+  disabled: boolean;
+  onClick: Function;
+  text: string;
+}
+
+interface StatesButtonState {
+  success: boolean;
+}
+
+export class StatesButton extends React.Component<
+  StatesButtonProps,
+  StatesButtonState
+> {
+  public buttonRef = React.createRef<HTMLButtonElement>();
+
+  public state = {
+    success: false,
+  };
+
+  public focus = () => {
+    this.buttonRef.current.focus();
+  };
+
+  public async onActionSuccess() {
+    this.setState({ success: true });
```

Comment:
We'll have to think of a different approach.
This won't work with the "loading..." state.